tests/js/session-tests.js

```diff
@@ -189,6 +189,22 @@ module.exports = {
     });
   },
 
+  async testRealmInvalidSyncConfiguration3() {
+    // test if an invalid object is used as user
+    const partition = Utils.genPartition();
+    let credentials = Realm.Credentials.anonymous();
+    let app = new Realm.App(appConfig);
+    let user = app.logIn(credentials);
+    let config = getSyncConfiguration(user, partition);
+    config.sync.user = { username: "John Doe" }; // this is an invalid user object
+    try {
+      await Realm.open(config);
+      Promise.reject();
+    } catch (_) {
```

Comment:
I would like this to assert something about the error message.
This will make it easier to detect and debug any other errors thrown from the call to `Realm.open`.